### ISSUE TITLE
antivirus: no header added

### ISSUE THREAD

**Issue by bax-** [vstakhov/rspamd#1620]
### Classification (Please choose *one* option):

* [ ] Crash/Hang/Data loss
* [ ] WebUI/Usability
* [ ] Serious bug
* [x] Other bug
* [ ] Feature
* [ ] Enhancement

### Reproducibility (Please choose *one* option):

* [X] Always
* [ ] Sometimes
* [ ] Rarely
* [ ] Unable
* [ ] I didn’t try
* [ ] Not applicable

### Rspamd version:
1.5.7
### Operation system, CPU, memory and environment:
FreeBSD 11
### Description (Please provide a descriptive summary of the issue):
ClamAV finds virus but no header is being added
### Steps to reproduce:

I send a Eicar-Test-Signature to myseld.

### Expected results:

X-Virus header added to Email.

### Actual results:

 Clamav finds it. But no header is being added – not even the X-Spamd-Resultthat normally should be there. 

### Configuration:

`local.d/antivirus.conf`
```
# multiple scanners could be checked, for each we create a configuration block with an arbitrary name
first {
  # If set force this action if any virus is found (default unset: no action is forced)
  # action = "reject";
  # if `true` only messages with non-image attachments will be checked (default true)
  attachments_only = true;
  # If `max_size` is set, messages > n bytes in size are not scanned
  #max_size = 20000000;
  # symbol to add (add it to metric if you want non-zero weight)
  symbol = "CLAM_VIRUS";
  # type of scanner: "clamav", "fprot", "sophos" or "savapi"
  type = "clamav";
  # If set true, log message is emitted for clean messages
  #log_clean = false;
  # For "savapi" you must also specify the following variable
  #product_id = 12345;
  # For "savapi" you can enable logging for clean messages
  #log_clean = true;
  # servers to query (if port is unspecified, scanner-specific default is used)
  # can be specified multiple times to pool servers
  # can be set to a path to a unix socket
  #servers = "127.0.0.1:3310";
  servers =  "/var/run/clamav/clamd.sock"
  # if `patterns` is specified virus name will be matched against provided regexes and the related
  # symbol will be yielded if a match is found. If no match is found, default symbol is yielded.
  patterns {
    # symbol_name = "pattern";
    JUST_EICAR = "^Eicar-Test-Signature$";
  }
  # `whitelist` points to a map of IP addresses. Mail from these addresses is not scanned.
  #whitelist = "/etc/rspamd/antivirus.wl";
}
```
`local.d/rmilter_headers.conf`
```
use = ["x-virus", "x-spamd-bar", "spam-header", "authentication-results"];

routines {
    x-spamd-bar {
      # effectively disables negative spambar
      neutral = "";
      negative = "";
    }
    x-virus {
        header = "X-Virus";
        remove = 1;
        # The following setting is an empty list by default and required to be set
        # These are user-defined symbols added by the antivirus module
        symbols = ["CLAM_VIRUS"];
    }
}
```

**Comment by fatalbanana** [comment]
Rmilter will (typically) add `X-Spamd-Result` header if metric result is `add header`.

You could check headers returned by rspamd using `rspamc --json`.

Perhaps the issue is that the message is awarded `JUST_EICAR` per configuration and you're looking merely for `CLAM_VIRUS`.

**Comment by bax-** [comment]
I have `Extended_spam_headers = yes;`and expected to always see the `X-Spamd-Result` header.
I just sent an emails to myself without the virus test file and there was also no extended header.
This might have something to do with sending locally then.

Using `action = "reject";` worked locally, as well as `action = "add header";`. In case of the latter, no `x-virus` header was added.

I can't send the test file from the other email addresses I have access to because the test file gets rejected on sending already.

I need to check if the headers are set properly and find out if I will reject such messages in the future or handle them via sieve.

**Comment by fatalbanana** [comment]
There are some [special rules governing rmilter's extended_spam_headers](https://rspamd.com/rmilter/configuration.html#spamd-section).

You don't really want to set action in antivirus module config, do you? In any case rmilter_headers should work irrespective (in my testing it seems it does).

**Comment by bax-** [comment]
Sorry, my bad. I missed the part with extended_spam_headers`.
Why wouldn't I want to set the action in antivirus module? What other options would I have and what would be the benefit?

I still don't understand why I don't get the `x-virus` header…

**Comment by bax-** [comment]
sorry – I think I get it.
I misunderstood the `patterns`part.

I added this to my `local.d/rmilter_headers.conf`:
```
symbols = ["CLAM_VIRUS", "JUST_EICAR"];
```

Now I got `X-Virus: Eicar-Test-Signature` in my mail header.
So everything seems to work.
What is the recommended way to deal with virus signatures in remitter then if antivirus module actions aren't the way to go?
Is there a way to put them in quarantine (postfix hold queue) like clamav-milter does?

thanks a lot!
